### PR TITLE
Fix: Type definition incorrect for resetContext

### LINKED ISSUE #309: Type definition incorrect for resetContext
## What are the steps to reproduce this issue?

The [documentation](https://docs.honeybadger.io/lib/javascript/guides/customizing-error-reports.html) provides the two following examples for `resetContext`

```
Honeybadger.resetContext();
// and
Honeybadger.resetContext({
  user_id: 123
});
```

But, type definition `honeybadger-js.d.ts` only covers the first example.

```
static resetContext(): Honeybadger;
```

## What happens?

This results in a Typescript error for the second example

## What were you expecting to happen?

No Typescript error

## Any logs, error output, etc?

```
[tsl] ERROR in /path/to/script
      TS2554: Expected 0 arguments, but got 1.
```

## Any other comments?

PR: https://github.com/honeybadger-io/honeybadger-js/pull/310

### PR DESCRIPTION
**Pull request by atshakil**
## Status

**READY**

## Description

resolves #309

### CODE REVIEW

File: honeybadger-js.d.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ declare module "honeybadger-js" {
         static onerror: boolean;
         static wrap<T extends Function>(func: T): T;
         static setContext<T extends Object>(context: T): Honeybadger;
-        static resetContext(): Honeybadger;
+        static resetContext(context?: T): Honeybadger;
```
Comment (anymaniax) [inline honeybadger-js.d.ts]: You forget to declare the generic like this `static resetContext<T extends Object>(context?: T): Honeybadger;`

**Comment by joshuap** [comment]
Thanks for putting this together! Let me know when it's ready and I'll merge/release.

**Comment by atshakil** [comment]
@anymaniax @joshuap I just updated. Please review.

**Comment by joshuap** [comment]
This has been released in v2.1.3

**Pull request merged**